### ISSUE TITLE
Input and output formats

### ISSUE THREAD

**Issue by scoavoux** [Lamarche-Perrin/triversity#2]
# Input formats

**tl;dr**: let's propose a method to input from a wide data frame or a set of data/dictionary data.frames

Currently, the input format of `get_multipartite()` should be a 4 or 5 columns data frame describing graph edges. However, many datasets in the social sciences will not be in this format, imho. They will consists either of:

+ a single wide data.frame with variables describing individuals, items and categories (the three node levels);
+ a set of data.frames, one with individuals/items, the other with items/categories.

Could we add built-in methods to use these? Here is a rough example draft for the first setup:

~~~r
get_tripartite_from_df <- function(df, lev1, lev2, lev3){
  require(magrittr)
  data.frame(V1 = c(rep(1, nrow(df)),
                    rep(2, nrow(df))),
             V2 = c(df[[lev1]],
                    df[[lev2]]),
             V3 = c(rep(2, nrow(df)),
                    rep(3, nrow(df))),
             V4 = c(df[[lev2]],
                    df[[lev3]])) %>% 
    get_multipartite(data = .)
}

get_tripartite_from_df(mydata, "user_id", "item_id", "category_id")

~~~

# Output formats

Currently, the output of  `get_diversity_from_path()` is a named numeric vector that can have two dimensions if several measures are chosen. It might be better to have a data.frame with a column containing (node labels) to facilitate merging the result with original data, which will be required if one wants to see the correlates of diversity. this would allow the use of `merge()` or `dplyr::*_join()`.

Current output:

~~~r
data("tripartite_example")
ex <- get_multipartite(data = tripartite_example)
str(get_diversity_from_path(ex, c(1, 2), measure = "entropy"))
~~~

~~~
num [1:2, 1] 1 2
 - attr(*, "dimnames")=List of 2
  ..$ : chr [1:2] "u1" "u2"
  ..$ : chr "entropy"
~~~

Proposed output:

~~~
'data.frame':	2 obs. of  2 variables:
 $ id     : chr  "u1" "u2"
 $ entropy: num  1 2
~~~

I can propose pull request with suggestions.

**Comment by Lamarche-Perrin** [comment]
Great, I am in favor of implementing both suggestions.

1. Regarding the input format, I am interested in the proposed implementation through a pull request.
2. Regarding the output format of `get_diversity_from_distribution`, and so of `get_diversity_from_path`, a former version of `triversity` was implementing the suggested output. Let me reintegrate it.
